tests/integration_tests/functional/test_rate_limiter.py

```diff
@@ -178,6 +177,13 @@ def test_rx_rate_limiting_cpu_load(test_microvm_with_api, network_config):
     )
     _iperf_out = _run_local_iperf(iperf_cmd)
 
+    # Stop the CPU monitor thread - otherwise it will crash after the microvm goes away due to no firecracker process
+    # existing anymore
+    test_microvm.cpu_load_monitor.signal_stop()
```

Comment:
hmm the `microvm` fixture should kill the vm after it is done. It also does the sample check. Is it not doing it?

Reply:
Yeah, I just saw that after Sudan's comment - that's an oversight on my part. However, it seems that the microvm dies before the sampler thread (because inside of `get_cpu_count` we now get an AssertionError about no `top` output being found - but this also happens right now if you look at the CI). I'll investigate this more later (will look at how the memory monitor does it and copy that approach), I'll put the PR back into draft mode for the time being. 

Aside,  do we really want to suppress exceptions on other threads? Was this a deliberate change, or should I also look into how to can revert that?

Reply:
Yeah if there is an easy way for pytest not to ignore those exceptions I think we would want that. Otherwise I fear it may be hiding bugs.

Reply:
Seems there is https://docs.pytest.org/en/6.2.x/usage.html#warning-about-unraisable-exceptions-and-unhandled-thread-exceptions

Reply:
Amazing, thanks, I'll tag a commit with that onto this PR then :)